ops/jenkins/bfd-build-deploy/Jenkinsfile

```
@@ -0,0 +1,185 @@
+        defaultValue: false
+        )
+
+        string(
+        name: 'server_regression_image_override',
+        description: 'Overrides the Docker image tag used when deploying the server-regression lambda',
+        defaultValue: null
+        )
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        script {
+            // Address limitations resulting from CVE-2022-24767
+            sh 'git config --global --add safe.directory "$WORKSPACE"'
+
+            // Get the current commit id
+            gitCommitId = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+
+            // Get the remote repo url. This assumes we are using git+https not git+ssh.
+            gitRepoUrl = sh(returnStdout: true, script: 'git config --get remote.origin.url').trim().replaceAll(/\.git$/, '')
+
+            // Load the child Jenkinsfiles.
+            scriptForApps = load('apps/build.groovy')
+            scriptForDeploys = load('ops/deploy-ccs.groovy')
+        }
+      }
+    }
+
+      /* This stage switches the gitBranchName (needed for our CCS downsream stages)
+          value if the build is a PR as the BRANCH_NAME var is populated with the build
+          name during PR builds.
+      */
+      stage('Set Branch Name') {
+        steps {
+          script {
+            currentStage = env.STAGE_NAME
+            // Set git branch name to master if not building amis
+            if (!params.force_build_amis) {
+              gitBranchName = 'master'
+            } else {
+              if (env.BRANCH_NAME.startsWith('PR')) {
+                gitBranchName = env.CHANGE_BRANCH
+              } else {
+                gitBranchName = env.BRANCH_NAME
+            }
+          }
+        }
+      }
+    }
+
+    stage('Run Build Apps') {
+      steps {
+        script {
+            if (params.force_build_amis) {
+              def jobToRun = "bfd-build-apps/${env.BRANCH_NAME.replace('/', '%2F')}"
+
+              build job:
+                jobToRun,
+                parameters: [
+                  booleanParam(name: "verbose_mvn_logging", value: "${params.verbose_mvn_logging}")
+                ],
+                wait: true
+          } else {
+            org.jenkinsci.plugins.pipeline.modeldefinition.Utils.markStageSkippedForConditional('Run Build Apps')
+          }
+        }
+      }
+    }
+
+    stage('Run Deploy Apps') {
+      steps {
+        script {
+          def jobToRun = "bfd-deploy-apps/${env.BRANCH_NAME.replace('/', '%2F')}"
+
+          awsAuth.assumeRole()
+
+          if (params.migrator_ami_override.length() == 0 || params.server_ami_override.length() == 0 || pipeline_ami_override.length() == 0) {
+            // Find the most current AMI IDs (if any).
+            amiIds = null
+            amiIds = scriptForDeploys.findAmis(gitBranchName)
+            migratorAmiId = amiIds.bfdMigratorAmiId
+            pipelineAmiId = amiIds.bfdPipelineAmiId
+            server_ami_override = amiIds.bfdServerAmiId
```

Let's keep the naming convention of our vars consistent if possible
```suggestion
            serverAmiId = amiIds.bfdServerAmiId
```